hack/tools/tilt-prepare/main.go

```diff
@@ -227,41 +234,115 @@ func tiltTools(ctx context.Context) error {
 func tiltResources(ctx context.Context, ts *tiltSettings) error {
 	tasks := map[string]taskFunction{}
 
+	// Note: we are creating clusterctl CRDs using kustomize (vs using clusterctl) because we want to create
+	// a dependency between these resources and provider resources.
+	tasks["clusterctl.crd"] = kustomizeTask("./cmd/clusterctl/config/crd/", "clusterctl.crd.yaml")
+
 	// If required, all the task to install cert manager.
 	// NOTE: strictly speaking cert-manager is not a resource, however it is a dependency for most of the actual resources
 	// and running this is the same task group of the kustomize/provider tasks gives the maximum benefits in terms of reducing the total elapsed time.
-	if *certManagerFlag {
+	if ts.DeployCertManager {
 		tasks["cert-manager-cainjector"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-cainjector:%s", config.CertManagerDefaultVersion))
 		tasks["cert-manager-webhook"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-webhook:%s", config.CertManagerDefaultVersion))
 		tasks["cert-manager-controller"] = preLoadImageTask(fmt.Sprintf("quay.io/jetstack/cert-manager-controller:%s", config.CertManagerDefaultVersion))
 		tasks["cert-manager"] = certManagerTask()
 	}
 
-	// Add a kustomize task for each name/path defined using the --kustomize-build flag.
-	for _, k := range *kustomizeBuildsFlag {
-		values := strings.Split(k, ":")
-		if len(values) != 2 {
-			return errors.Errorf("[resources] failed to parse --kustomize-build flag %s: value should be in the form of name:path", k)
-		}
-		name := values[0]
-		path := values[1]
+	// Add a kustomize task for each tool configured via deploy_observability.
+	for _, tool := range ts.DeployObservability {
+		name := fmt.Sprintf("%s.observability", tool)
+		path := fmt.Sprintf("./hack/observability/%s/", tool)
 		tasks[name] = kustomizeTask(path, fmt.Sprintf("%s.yaml", name))
 	}
 
-	// Add a provider task for each name/path defined using the --provider flag.
-	for _, p := range *providersBuildsFlag {
-		pValues := strings.Split(p, ":")
-		if len(pValues) != 2 {
-			return errors.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path", p)
+	providerPaths := map[string]string{"core": ".",
+		"kubeadm-bootstrap":     "bootstrap/kubeadm",
+		"kubeadm-control-plane": "controlplane/kubeadm",
+		"docker":                "test/infrastructure/docker",
+	}
+
+	// Add all the provider paths to the providerpaths map
+	for _, p := range ts.ProviderRepos {
+		providerContexts, err := loadProviders(p)
+		if err != nil {
+			return err
+		}
+
+		for name, path := range providerContexts {
+			providerPaths[name] = path
+		}
+	}
+
+	ts.EnableProviders = append(ts.EnableProviders, "core")
```

Comment:
nit, core should be added only if not already there (otherwise we add it twice)

Reply:
added a condition